web/templates/web/demographic-data-update.html

```diff
@@ -12,25 +12,89 @@
 {% block content %}
     <script>
         $(document).ready(function() {
-            if ($('#id_country')[0].value !== 'US') {
-                hideField('state');
-            }
+        
+			const gdpr_country_list = [	
+				"", // don't accept demographic data with NO country listed
+				"AT", 
+				"BE", 
+				"BG", 
+				"CY", 
+				"CZ", 
+				"DE", 
+				"DK", 
+				"EE", 
+				"ES", 
+				"FI", 
+				"FR", 
+				"GB", 
+				"GR", 
+				"HR", 
+				"HU", 
+				"IE", 
+				"IT", 
+				"LT", 
+				"LU", 
+				"LV", 
+				"MT", 
+				"NL", 
+				"PL", 
+				"PT", 
+				"RO", 
+				"SE", 
+				"SI", 
+				"SK",
+				"NO", // protections also apply to Norway, Iceland, & Liechtenstein
+				"IS",
+				"LI"];
+			
+			const gdprFieldsToHide = ['number_of_children', 'child_birthdays', 
+            		'languages_spoken_at_home', 'number_of_guardians', 
+            		'number_of_guardians_explanation', 'race_identification', 
+            		'age', 'gender', 'education_level', 'spouse_education_level',
+            		'annual_income', 'number_of_books', 'density'];
+        
             if ($('#id_number_of_children')[0].value === "0" || $('#id_number_of_children')[0].value === '') {
                 hideField('child_birthdays');
                 $('.help-block').first().hide();
             }
             if ($('#id_number_of_guardians')[0].value !== 'varies') {
                 hideField('number_of_guardians_explanation');
             }
-
+            
+            if ($('#id_country')[0].value !== 'US') {
+                hideField('state');
+            }
+            
+            if (gdpr_country_list.includes($('#id_country')[0].value)) {
+            	$.each(gdprFieldsToHide, function(i, fieldName) {
+            		hideField(fieldName);
+            		$(`#id_${fieldName}`)[0].value = '';
+            		$(`#id_${fieldName} input[type=checkbox]`).prop("checked", false );
+            	});
+            }
+            
             $('#id_country').on('change', function() {
+            	// First show/hide state field based on USA or not
```

Comment:
Just for my understanding, why do you have to show/hide state field first?

Reply:
Sorry, bad comment. What I meant was just do state fields and everything else *separately* for clarity. (It was briefly structured as "are you in the US? if show, show state. If not, are you in the EU? Show/hide personal data fields accordingly." so that fields weren't appropriately showing back up if someone selected US.)